src/mo_frontend/test_recovery.ml

```diff
@@ -84,9 +96,123 @@ let%expect_test "test1" =
     )
 
      with errors:
-    (unknown location): syntax error [M0001], unexpected token 'let'
-
-    (unknown location): syntax error [M0001], unexpected token 'let' |}]
+    (unknown location): syntax error [M0001], unexpected token 'let', expected one of token or <phrase> sequence:
+      } (e.g. '}')
+      .<nat> (e.g. '.<nat>')
+      ! (e.g. '!')
+      <exp_nullary(ob)> (e.g. '42')
+      <unop> <exp_bin(ob)> (e.g. '^ 42')
+      <unassign> <exp(ob)> (e.g. '^= 42')
+      <binop> <exp_bin(ob)> (e.g. '-% 42')
+      <binassign> <exp(ob)> (e.g. '-%= 42')
+      <binop> <exp_bin(ob)> (e.g. '**% 42')
+      <binassign> <exp(ob)> (e.g. '**%= 42')
+      <binop> <exp_bin(ob)> (e.g. '*% 42')
```

Comment:
For example:
```
      <binop> <exp_bin(ob)> (e.g. '-% 42')
      <binassign> <exp(ob)> (e.g. '-%= 42')
      <binop> <exp_bin(ob)> (e.g. '**% 42')
      <binassign> <exp(ob)> (e.g. '**%= 42')
      <binop> <exp_bin(ob)> (e.g. '*% 42')
```
could be condensed to
```
      <binop> <exp_bin(ob)> (e.g. '+ 42')
      <binassign> <exp(ob)> (e.g. '+= 42')
```
Since the lhs are repeated but the rhs are just minor variations.

Reply:
fixed